cmd/frontend/graphqlbackend/external_services.go

```
@@ -49,8 +53,16 @@ func (r *schemaResolver) AddExternalService(ctx context.Context, args *struct {
 		DisplayName: args.Input.DisplayName,
 		Config:      args.Input.Config,
 	}
-	err := db.ExternalServices.Create(ctx, externalService)
-	return &externalServiceResolver{externalService: externalService}, err
+
+	if err := db.ExternalServices.Create(ctx, externalService); err != nil {
+		return nil, err
+	}
+
+	if err := syncExternalService(ctx, externalService); err != nil {
+		return nil, err
```

The external service has been created at this point, so I am wondering if returning an error here is the right thing to do. If you do want to return an error here, it should probably be more descriptive "External service was created but ....."

---

Good call. I'll make the error more descriptive.